spec-dtslint/operators/pluck-spec.ts

```diff
@@ -41,6 +41,10 @@ it('should accept string only', () => {
   const a = of({ name: 'abc' }).pipe(pluck(1)); // $ExpectError
 });
 
+it('should not infer type from the variable if key doesn\'t exist', () => {
+  const a: Observable<number> = of({ name: 'abc' }).pipe(pluck('xyz')); // $ExpectError
```

Comment:
```suggestion
  const a: Observable<never> = of({ name: 'abc' }).pipe(pluck('xyz'));
```

Wouldn't that be a better test?

Reply:
No, this test is meant to show that predefined type doesn't match to the `unknown` that is calculated from the `pluck` when the key doesn't exist.

In current implementation this test will fail. The `pluck('xyz')` return type is `number` and the key doesn't exist. It should be `unknown` but it's inferring from the predefined type of the variable/constant.

Reply:
This test should be a positive failure, and the correction you are suggesting has already been checked in `should accept existing keys only` test case.

Consider this example:

```ts
interface SomeObject {
    a: string;
}

interface Input {
    b: SomeObject;
}

interface Output {
    b: Observable<SomeObject>;
}

const stream$ = of<Input>({
    b: { a: 'abc' },
});

const positiveCase$ = stream$.pipe(pluck('b')); // will be of type Observable<SomeObject>

const negativeCase$ = stream$.pipe(pluck('xyz')); // should return Observable<unknown> but the current implementation will infer the type and the type will be Observable<SomeObject>

const positiveOutput: Output = {
    b: positiveCase$, // no error here
}

const negativeOutput: Output = {
    b: negativeCase$, // will throw type error with my implementation and will be silent using current one
}
```